src/components/Cluster/NewCluster/CreateRegularCluster.js

```diff
@@ -408,213 +380,177 @@ class CreateRegularCluster extends React.Component {
-                            />
-                            <p>{`${CPUCores} CPU cores, ${RAM} GB RAM each`}</p>
-                          </FlexWrapperDiv>
-                        </>
-                      );
-                    }
-                  }
-
-                  return null;
-                })()}
-              </label>
-              <label className='scaling-range' htmlFor='scaling-range'>
-                <span className='label-span'>Number of worker nodes</span>
-                <NodeCountSelector
-                  autoscalingEnabled={CreateRegularCluster.isScalingAutomatic(
-                    provider,
-                    this.props.selectedRelease
-                  )}
-                  maxValue={this.props.maxWorkersPerCluster}
-                  onChange={this.updateScaling}
-                  readOnly={false}
-                  scaling={this.state.scaling}
-                />
-                <ProviderCredentials
-                  organizationName={this.props.selectedOrganization}
-                  provider={provider}
-                />
-                {this.state.error && this.errorState()}
-              </label>
-            </FlexColumnDiv>
-
-            <hr style={{ margin: '37px 0 31px' }} />
-
-            <FlexRowDiv>
-              <ErrorFallbackStyled error={noReleasesErrorText} />
-              <ErrorFallback error={this.props.clusterCreateError}>
-                <Button
-                  bsSize='large'
-                  bsStyle='primary'
-                  disabled={!this.valid()}
-                  loading={this.state.submitting}
-                  onClick={this.createCluster}
-                  type='submit'
-                >
-                  Create Cluster
-                </Button>
-              </ErrorFallback>
-              {!this.state.submitting && (
-                <Button
-                  bsSize='large'
-                  bsStyle='default'
-                  loading={this.state.submitting}
-                  onClick={this.props.closeForm}
-                  type='button'
-                >
-                  Cancel
-                </Button>
-              )}
-            </FlexRowDiv>
-            <FlexColumnDiv>
-              <ClusterCreationDuration
-                stats={this.props.clusterCreationStats}
+      <WrapperDiv data-testid='cluster-creation-view'>
+        <HorizontalLine />
+
+        <FlexColumn>
+          <div className='worker-nodes'>Worker nodes</div>
+          {(provider === Providers.AWS || provider === Providers.AZURE) && (
+            <V4AvailabilityZonesSelector
+              minValue={this.props.minAvailabilityZones}
+              maxValue={this.props.maxAvailabilityZones}
+              onChange={this.updateAvailabilityZonesPicker}
+              {...multiAZSelectorProps}
+            />
+          )}
+          <label htmlFor='instance-type' className='instance-type'>
+            {(() => {
```

Comment:
Would be awesome if we could move this switch statement into a function, and extract each switch case to a separate component. What do you think?

Reply:
Agree, I can do it in the same PR where I do this: https://github.com/giantswarm/happa/pull/1544#discussion_r437637115